### PR TITLE
feat: farcaster login

### PR DESCRIPTION
**Pull request by gregfromstl**
### TL;DR

Added Farcaster icon and login support to the thirdweb package.

### What changed?

- Added Farcaster icon to `socialIcons.ts` and `svgs.ts`
- Updated `WalletImage.tsx` to include Farcaster icon.
- Added Farcaster as a supported social login option throughout the codebase.

### How to test?

1. Ensure the Farcaster icon is rendered correctly by using the Farcaster login option.
2. Verify the Farcaster login flow works as expected on both web and native applications.

### Why make this change?

This update adds support for Farcaster as a social login option, enhancing the flexibility and options available to users for authentication.

---

---

## PR-Codex overview
This PR adds support for the "Farcaster" authentication option in the in-app wallets feature. It includes UI updates, authentication logic, and social icons.

### Detailed summary
- Added "Farcaster" as an authentication option in various files
- Updated authentication logic for "Farcaster" in different connectors
- Included "Farcaster" in social authentication options and UI components
- Defined "Farcaster" icon URI for web and native UI components

> The following files were skipped due to too many changes: `packages/thirdweb/src/react/core/utils/socialIcons.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by gregfromstl** [comment]
* **#3827** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/3827?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gregfromstl and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .changeset/spicy-flowers-return.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+"thirdweb": minor
+---
+
+Adds SIWF for in-app wallets
+
+```ts
+await wallet.connect({
+    strategy: "farcaster",
```
Comment (joaquim-verges) [inline .changeset/spicy-flowers-return.md:9]: pog

File: apps/playground-web/src/lib/constants.ts [hunk 2]
```diff
@@ -9,12 +9,12 @@ export const WALLETS = [
     auth: {
       options: [
         "google",
-        "facebook",
```
Comment (jnsdls) [inline apps/playground-web/src/lib/constants.ts:12]: wait, w're exchanging fb for fc?

File: packages/thirdweb/src/react/native/ui/connect/InAppWalletUI.tsx [hunk 3]
```diff
@@ -62,9 +68,8 @@ export function InAppWalletUI(props: InAppWalletFormUIProps) {
   const { wallet, theme } = props;
   const config = wallet.getConfig();
   const authOptions = config?.auth?.options || defaultAuthOptions;
-  const socialLogins = authOptions.filter(
-    (x) =>
-      x === "google" || x === "apple" || x === "facebook" || x === "discord",
+  const socialLogins = authOptions.filter((x) =>
+    socialAuthOptions.includes(x as SocialAuthOption),
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/connect/InAppWalletUI.tsx:72]: weird is this `as` necessary? why? shouldn't typescript 5.5 handle this?

File: packages/thirdweb/src/react/native/ui/icons/svgs.ts [hunk 4]
```diff
@@ -37,6 +37,13 @@ export const DISCORD_ICON = `<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink
 </g>
 </svg>`;
 
+export const FARCASTER_ICON = `<svg width={width} height={height} viewBox="0 0 1000 1000" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (jnsdls) [inline packages/thirdweb/src/react/native/ui/icons/svgs.ts:40]: wait so we have SVG and base64 of (assuringly) the same svg? - can we simplify this somehow? (not new to this but I just realized it)

File: packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx [hunk 5]
```diff
@@ -91,6 +91,7 @@ export const ConnectWalletSocialOptions = (
     facebook: locale.signInWithFacebook,
     apple: locale.signInWithApple,
     discord: locale.signInWithDiscord,
+    farcaster: "Farcaster",
```
Comment (jnsdls) [inline packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx:94]: I guess can't localize this anyways?

**Review by jnsdls — APPROVED** [review]
left some questions and NITs but overall LGTM

File: packages/thirdweb/src/react/native/ui/connect/InAppWalletUI.tsx (hunk 3 above)
Comment (gregfromstl) [reply packages/thirdweb/src/react/native/ui/connect/InAppWalletUI.tsx:72]: Yeah been seeing this a lot, haven't dug into why

File: packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx (hunk 5 above)
Comment (gregfromstl) [reply packages/thirdweb/src/react/web/wallets/shared/ConnectWalletSocialOptions.tsx:94]: Yeah the locales are just the names anyway, feels unnecessary

File: apps/playground-web/src/lib/constants.ts (hunk 2 above)
Comment (gregfromstl) [reply apps/playground-web/src/lib/constants.ts:12]: On the playground only

Comment (jnsdls) [reply apps/playground-web/src/lib/constants.ts:12]: u right

**Pull request merged**